<!DOCTYPE HTML>

<html>
	<head>
		<title>Unsupervised clustering on scRNAseq cardiac stem cell data</title>
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no" />
		<link rel="stylesheet" href="assets/css/main.css" />
	</head>
	<body class="is-preload">

		<!-- Wrapper -->
			<div id="wrapper">

				<!-- Main -->
					<div id="main">
						<div class="inner">

							<!-- Header -->
							<header id="header">
								<a href="index.html" class="logo"><strong>Project 2: Data analysis</strong> by Ferran Cardoso</a>
								<ul class="icons">
									<li><a href="https://github.com/FerranC96/MSc-Project_2" class="icon fa-github"><span class="label">Github</span></a></li>
									<li><a href="mailto:ferran.cardoso-rodriguez18@imperial.ac.uk" class="icon fa-envelope-o"><span class="label">Email</span></a></li>
								</ul>
							</header>

							<!-- Content -->
							<section>
									<header class="main">
										<h1>2.- Aims and initial data</h1>
									</header>

									<hr class="major" />

									<h2>Aim</h2>
									<p>The aim of this project is to characterise subpopulations within the PDGFRα positive adult cardiac murine cells via unsupervised clustering techniques, in an effort to gain insights into what might drive their cardiogenic potential.</p>
									<p>To this end clustering techniques allow us to identify putative subpopulations within our dataset. Subsequent Differential Expression assays tell us which, if any, clusters are characterised by marker genes, wich will then be used for a biological assessment.</p>

									<hr class="major" />
									<div class="row">
										<div class="col-6 col-12-small">
											
											<h2>Initial Data</h2>
											
										</div>
										<div class="col-6 col-12-small">
											<p align="right"><a href="data/pdgfra_reads.7z" class="button icon fa-download">Download raw data</a></p>
											</div>
									</div>
									<p>The data for the project was, as introduced in the previous section, acquired by performing single cell RNA sequencing (scRNAseq) on the PDGFRα positive population that had been selected via FACS. After the full length scRNAseq with Smart-Seq2 (<a href="#">Picelli et al., 2013</a>) the data was initiallly pre-processed and aligned to the murine transcriptome by Dr. Andrea Massaia at Noseda’s group. Further details on how the data was obtained and handled can be found on <a href="#">Massaia et al., 2018</a>.</p>
									<p>Raw data, which can be downloaded from the button above, consists of an expression matrix with columns representing cells and rows being <em>Ensembl</em> transcript identifiers, with 113.779 rows and 416 columns in total. Spike-in data from the commercial mix ERCC (External RNA Controls Consortium) was added during the sequencing to use as a control in the pre-processing steps outlined in the <a href="intro.html">following section</a>. </p>
									<p>While the expression matrix can be loaded into R directly as a data frame, we converted it into a Single Cell Experiment object (SCE) using its homonymous package in R (<a href="#">Lun A, Risso D, 2019</a>), which has been specifically designed to handle, store and retrieve single cell experiment data. Its usage, in addition to the quality control and visualisation  package <em>scater</em> (<a href="#">McCarthy et al., 2017</a>) will form the backbone of the analysis presented here.
										As the SCE object was being created, the ERCC spike in information was stored into the SCE object to be used during quality control (see <a href="pre-process.html#qc">QC and filtering</a>) as method to detect poor quality cells. In our data we find that, from a total of 113779 transcript entries, 92 are in fact labelled as ERCC spike-ins. More information on the use of ERCC for quality control purposes can be found on section XXXX.</p>
									<hr class="major" />
									<p align="right"><a href="intro.html" class="button primary">Next</a></p>

								</section>

						</div>
					</div>

				<!-- Sidebar -->
					<div id="sidebar">
						<div class="inner">

							<!-- Menu -->
							<nav id="menu">
									<header class="major">
										<h2>Menu</h2>
									</header>
									<ul>
										<li><a href="index.html">Abstract</a></li>
										<li><a href="background.html">1.- Background</a></li>
										<li><a href="aim.html">2.- Aims and Initial Data</a></li>
										<li><a href="intro.html">3.- Introduction and pipeline</a></li>
										<li>
											<span class="opener">4.- Pre-processing</span>
											<ul>
												<li><a href="pre-process.html">4.1- QC and filtering</a></li>
												<li><a href="pre-process.html#norm">4.2- Normalisation</a></li>
											</ul>
										</li>
										<li>
											<span class="opener">5.- Visualisation and Clustering</span>
											<ul>
												<li><a href="analysis.html">5.1- Dimensionality Reduction</a></li>
												<li><a href="analysis.html#clust">5.2- Unsupervised clustering</a></li>
												<li><a href="analysis.html#bio">5.3- Biological analysis</a></li>
											</ul>
										</li>
										<li>
											<span class="opener">6.-Discussion</span>
											<ul>
												<li><a href="discussion.html">6.1- Discussion</a></li>
												<li><a href="discussion.html#conclussion">6.2- Conclusions</a></li>
												<li><a href="discussion.html#future">6.3- Future work</a></li>
											</ul>
										</li>
										<li><a href="refs.html">7.-References</a></li>
										<li><a href="elements.html">Elements</a></li>
									</ul>
								</nav>

							<!-- Section -->
								<!-- <section>
									<header class="major">
										<h2>Ante interdum</h2>
									</header>
									<div class="mini-posts">
										<article>
											<a href="#" class="image"><img src="images/pic07.jpg" alt="" /></a>
											<p>Aenean ornare velit lacus, ac varius enim lorem ullamcorper dolore aliquam.</p>
										</article>
										<article>
											<a href="#" class="image"><img src="images/pic08.jpg" alt="" /></a>
											<p>Aenean ornare velit lacus, ac varius enim lorem ullamcorper dolore aliquam.</p>
										</article>
										<article>
											<a href="#" class="image"><img src="images/pic09.jpg" alt="" /></a>
											<p>Aenean ornare velit lacus, ac varius enim lorem ullamcorper dolore aliquam.</p>
										</article>
									</div>
									<ul class="actions">
										<li><a href="#" class="button">More</a></li>
									</ul>
								</section> -->

							<!-- Section -->
							<section>
								<header class="major">
									<h2>Acknowledgements</h2>
								</header>
								<p>I would like to thank Prof. Mauricio Barahona and Dr. Michela Noseda for their supervision and guidance throughout the project.<br />Special thanks as well to Andrea Massaia, PhD, for all his support and scientific insights that made this project possible. </p>
	
							</section>

						<!-- Footer -->
							<footer id="footer">
								<p class="copyright">&copy; Ferran Cardoso.<br />Project 2: Data Analysis. <br />MSc in Bioinformatics and Theoretical Systems Biology.<br />Imperial College London. <br /> Design: <a href="https://html5up.net">HTML5 UP</a>.</p>
							</footer>

						</div>
					</div>

			</div>

		<!-- Scripts -->
			<script src="assets/js/jquery.min.js"></script>
			<script src="assets/js/browser.min.js"></script>
			<script src="assets/js/breakpoints.min.js"></script>
			<script src="assets/js/util.js"></script>
			<script src="assets/js/main.js"></script>

	</body>
</html>