<!DOCTYPE HTML>

<html>
	<head>
		<title>Unsupervised clustering on scRNAseq cardiac stem cell data</title>
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no" />
		<link rel="stylesheet" href="assets/css/main.css" />
	</head>
	<body class="is-preload">

		<!-- Wrapper -->
			<div id="wrapper">

				<!-- Main -->
					<div id="main">
						<div class="inner">

							<!-- Header -->
							<header id="header">
								<a href="index.html" class="logo"><strong>Project 2: Data analysis</strong> by Ferran Cardoso</a>
								<ul class="icons">
									<li><a href="https://github.com/FerranC96/MSc-Project_2" class="icon fa-github"><span class="label">Github</span></a></li>
									<li><a href="mailto:ferran.cardoso-rodriguez18@imperial.ac.uk" class="icon fa-envelope-o"><span class="label">Email</span></a></li>
								</ul>
							</header>

							<!-- Content -->
								<section>
									<header class="main">
										<h1>5.- Visualisation and clustering</h1>
									</header>

									<hr class="major" />
                                    <h2 id="dr">5.1- Dimensionality Reduction</h2>
                                    <p>Data visualisation is an essential, yet sometimes overlooked, step of data analysis.  In our case Dimensionality Reduction (DR) techniques will prove useful both for visualisation and clustering purposes, as scRNAseq data contains a huge number of dimensions.
										DR will reduce the dimensions of the data to facilitate its manipulation and feeding into downstream clustering algorithms. </p>
                                    <p>In this section the different DR approaches taken are shown, with a brief description of each method used, its strength and weaknesses, and plots for the final DR spaces they produce. </p>
									
									<h3 id="pca">Principal Component Analysis</h3>                                    
                                    <div class="row">
                                            <div class="col-6 col-12-small">
												<p>Principal Component Analysis, or PCA, was the first approach taken. It is a transformation-based approach that calculates new dimensions or components to try and capture the maximum amount of variance possible in the original high-dimensional space.</p>
												<p>Making use of the technical variances calculated in the <a href="pre-process.html">previous section</a> we can remove that random noise by performing PCA and <em>scran</em>’s denoisePCA function. </p>
												<p>A plot of our cells in the PCA space is shown on <a href="#pca_plot">Figure 8</a>. It is immediately noticeable that no apparent division of cells appears, with all cells clustered together as one mass, 
													and that PCA does not render very informative results for the first two components explain only 3% and 1% of the total variance. The reasons this might be happening will be discussed in the <a href="discussion.html">following section</a>,
													but PCA has already been described as being a less than ideal DR approach for sparse scRNAseq data.</p>
                                            </div>
                                            <div class="col-6 col-12-small">
                                                <div class="box">
                                                <span class="image fit"><img src="images/dr_pca.png" id="pca_plot" /></span>
                                                <p>Figure 8: .</p>
                                                </div>
                                            </div>
									</div>
									
                                    <h3 id="tsne">t-Distributed Stochastic Neighbour Embedding</h3>                                    
                                    <div class="row">
                                            <div class="col-6 col-12-small">
												<p>t-Distributed Stochastic Neighbour Embedding, also known as t-SNE, is a newer DR approach described in <a href="#">Maaten and Hinton, 2008</a>. It maps high dimensional data to 2 dimensions by performing random walks on neighbourhood graphs and can deal better results in more diverse and higher dimensional cell populations.</p>
												<p>To make use of the denoising performed above we calculate its reduced dimensions from the PCA data. A stochastic process by definition, the resulting graphs all showed different topologies even if the perplexity value was kept constant.</p>
												<p>On <a href="#tsne_plot">Figure 9</a> one such plot is given with a perplexity value of 15 and no random seed predefined. I settled at perplexity=15 since that seemed to produce the most consistent topologies while still avoiding the tendency to crowd points together at the centre of the map.</p>
                                            </div>
                                            <div class="col-6 col-12-small">
                                                <div class="box">
                                                <span class="image fit"><img src="images/dr_tsne.png" id="tsne_plot" /></span>
                                                <p>Figure 9: .</p>
                                                </div>
                                            </div>
									</div>
									
									<h3 id="umap">Uniform Manifold Approximation and Projection</h3>                                    
                                    <div class="row">
                                            <div class="col-6 col-12-small">
												<p>Uniform Manifold Approximation and Projection (UMAP) is DR technique that can be used in similar situations as t-SNE. Published last year on <a href="#">McInnes et al., 2018</a> it works based on manifold modelling via fuzzy topologies and subsequent embedding on a lower-dimension projection.
													Its main advantages over t-SNE are that it is less computationally intense while better preserving global structure of the data at comparable details of local neighbour relations.</p>
												<p>UMAP plots obtained from our data presented more consistent overall topologies than their t-SNE counterparts, while adjusting to the clusters found better than PCA.  A UMAP plot can be seen on <a href="#umap_plot">Figure 10</a>.</p>
												
                                            </div>
                                            <div class="col-6 col-12-small">
                                                <div class="box">
                                                <span class="image fit"><img src="images/dr_umap.png" id="umap_plot" /></span>
                                                <p>Figure 10: .</p>
                                                </div>
                                            </div>
									</div>
																		
									<hr class="major" />
                                    <h2 id="clust">5.2- Unsupervised clustering</h2>
                                    <p>As different cells have different capture efficiencies and the sequencing depth is not completely homogenous throughout the sample, normalisation methods are necessary to palliate these cell-specific biases. 
                                        To that end I performed a CPM-based (Counts Per Million) method from <em>scran</em> that has been specifically tailored to deal with the sparse expression matrices characteristic of scRNAseq experiments. </p>
                                    <p>Before computing the size factors necessary for the normalisation, I performed a preliminary clustering step (based on hierarchical clustering) to palliate the consequences of the DE assumption central to <em>scran</em>’s normalisation method. 
                                        This increases robustness towards heterogeneity, and now size factors for both endogenous genes and spike-ins can be calculated before normalising the data and storing it as log-transformed normalised counts.</p>
                                    <p>Another confounding effect that remains to be removed are the technical effects, which can alter the observed expression values. To model this technical noise the ERCC spike in set was used, as its expression should remain relatively constant across cells
										and thus if there is no biological variability, any variance observed should then be caused by technical effects. This modelling consisted of variance fitting and was performed using the <em>trendVar()</em> and <em>decomposeVar()</em> functions from <em>scran</em>>.</p>
									
									<h3>t-SNE + k-means</h3>
									
									<h3>UMAP + k-means</h3>

									<h3>SC3</h3>
									
									<h3>Comparison</h3>

									<hr class="major" />
									<h2 id="bio">5.3- Biological Analysis</h2>


									
									
      
									<hr class="major" />
									<p align="right"><a href="analysis.html" class="button primary">Next</a></p>

								</section>

						</div>
					</div>

				<!-- Sidebar -->
				<div id="sidebar">
					<div class="inner">

						<!-- Menu -->
						<nav id="menu">
								<header class="major">
									<h2>Menu</h2>
								</header>
								<ul>
									<li><a href="index.html">Abstract</a></li>
									<li><a href="background.html">1.- Background</a></li>
									<li><a href="aim.html">2.- Aims and Initial Data</a></li>
									<li><a href="intro.html">3.- Introduction and pipeline</a></li>
									<li>
										<span class="opener">4.- Pre-processing</span>
										<ul>
											<li><a href="pre-process.html">4.1- QC and filtering</a></li>
											<li><a href="pre-process.html#norm">4.2- Normalisation</a></li>
										</ul>
									</li>
									<li>
										<span class="opener">5.- Visualisation and Clustering</span>
										<ul>
											<li><a href="analysis.html">5.1- Dimensionality Reduction</a></li>
											<li><a href="analysis.html#clust">5.2- Unsupervised clustering</a></li>
											<li><a href="analysis.html#bio">5.3- Biological analysis</a></li>
										</ul>
									</li>
									<li>
										<span class="opener">6.-Discussion</span>
										<ul>
											<li><a href="discussion.html">6.1- Discussion</a></li>
											<li><a href="discussion.html#conclussion">6.2- Conclusions</a></li>
											<li><a href="discussion.html#future">6.3- Future work</a></li>
										</ul>
									</li>
									<li><a href="refs.html">7.-References</a></li>
									<li><a href="elements.html">Elements</a></li>
								</ul>
							</nav>

						<!-- Section -->
							<!-- <section>
								<header class="major">
									<h2>Ante interdum</h2>
								</header>
								<div class="mini-posts">
									<article>
										<a href="#" class="image"><img src="images/pic07.jpg" alt="" /></a>
										<p>Aenean ornare velit lacus, ac varius enim lorem ullamcorper dolore aliquam.</p>
									</article>
									<article>
										<a href="#" class="image"><img src="images/pic08.jpg" alt="" /></a>
										<p>Aenean ornare velit lacus, ac varius enim lorem ullamcorper dolore aliquam.</p>
									</article>
									<article>
										<a href="#" class="image"><img src="images/pic09.jpg" alt="" /></a>
										<p>Aenean ornare velit lacus, ac varius enim lorem ullamcorper dolore aliquam.</p>
									</article>
								</div>
								<ul class="actions">
									<li><a href="#" class="button">More</a></li>
								</ul>
							</section> -->

						<!-- Section -->
						<section>
							<header class="major">
								<h2>Acknowledgements</h2>
							</header>
							<p>I would like to thank Prof. Mauricio Barahona and Dr. Michela Noseda for their supervision and guidance throughout the project.<br />Special thanks as well to Andrea Massaia, PhD, for all his support and scientific insights that made this project possible. </p>

						</section>

					<!-- Footer -->
						<footer id="footer">
							<p class="copyright">&copy; Ferran Cardoso.<br />Project 2: Data Analysis. <br />MSc in Bioinformatics and Theoretical Systems Biology.<br />Imperial College London. <br /> Design: <a href="https://html5up.net">HTML5 UP</a>.</p>
						</footer>

					</div>
				</div>

			</div>

		<!-- Scripts -->
			<script src="assets/js/jquery.min.js"></script>
			<script src="assets/js/browser.min.js"></script>
			<script src="assets/js/breakpoints.min.js"></script>
			<script src="assets/js/util.js"></script>
			<script src="assets/js/main.js"></script>

	</body>
</html>