<!DOCTYPE HTML>

<html>
	<head>
		<title>Unsupervised clustering on scRNAseq cardiac stem cell data</title>
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no" />
		<link rel="stylesheet" href="assets/css/main.css" />
	</head>
	<body class="is-preload">

		<!-- Wrapper -->
			<div id="wrapper">

				<!-- Main -->
					<div id="main">
						<div class="inner">

							<!-- Header -->
							<header id="header">
								<a href="index.html" class="logo"><strong>Project 2: Data analysis</strong> by Ferran Cardoso</a>
								<ul class="icons">
									<li><a href="https://github.com/FerranC96/MSc-Project_2" class="icon fa-github"><span class="label">Github</span></a></li>
									<li><a href="mailto:ferran.cardoso-rodriguez18@imperial.ac.uk" class="icon fa-envelope-o"><span class="label">Email</span></a></li>
								</ul>
							</header>

							<!-- Content -->
								<section>
									<header class="main">
										<h1>6.- Discussion</h1>
									</header>

									<hr class="major" />

									<h2>6.1- Discussion</h2>
                                    <h3>Pre-processing</h3>
                                    <p>Apparent from the first steps of QC and filtering was the fact that this data had already undergone a selection process and was most certainly part of a bigger expression matrix containing a similar or identical number of rows (transcript IDs) but a much greater number of columns (cells).
                                        Of those, if only high-quality cells identified as PDGFRα+ve during flow cytometry where selected this would explain why, as shown on <a href="pre-process.html#qc">Section 4.1</a>, cell-based metrics >resulted in orders of magnitude less filtered entries than gene-based metrics
                                        (on which the inherent sparsity of scRNAseq data is accentuated by leaving in transcripts that are no longer expressed in any of the cells). </p>
                                    <p>It is because of that observation that a gene filtering step with M3Drop (<a href="#">Andrews T, 2019</a>) was considered, since drop-out selection of irrelevant genes would clean such matrices very efficiently. However, I was afraid I could loose precious information from an already quite homogeneous population so in the end it wasn’t incorporated into the pipeline.</p>
                                    
                                    <h3>Visualisation and clustering</h3>
                                    <p>Plotting the data on the different dimensionally reduced spaces obtained in <a href="analysis.html#dr">Section 5.1</a> did not seem to give us any easily identifiable subpopulations, with only t-SNE and UMAP showing topologies that could suggest there could be some clusters in the data.
                                        Applying the different clusters found in <a href="analysis.html#clust">Section 5.2</a> did not have really satisfactory results when k was low, but at higher k it appears that the clusters where projecting better on the reduced space,
                                        possibly due to the finer resolution being able to capture smaller number of cells that better mimic topological features in the DR spaces.</p>
                                    <p>This phenomenon brings us to the criteria for selecting a suitable number of clusters. Aiming for a compromise between a good enough silhouette (a measure of how robust the clusters obtained are)
                                        and a sensible number of clusters based on the biology of our study population, I ended settling at k=4, since that would give a silhouette greater than 0.65 while still projecting adequately on the UMAP plot. Biologically it is also more feasible to have 4 different subpopulations in our data, rather than 9. </p>
                                    
                                    <h3>Biological analysis</h3>
                                    <p>Regarding the markers found on <a href="analysis.html#bio">Section 5.3</a>, two observations appeared immediately noticeable. The first one was that the markers identified with cluster 1 are also very descriptive of cluster 2, even if they do not reach statistical significance.</p>
                                    <p>The other two markers characteristic of cluster 4 are more contained in expression, AY036118 showing the highest DE of all markers found. It is also this marker that shows seems the most interesting in the biological sense, for it was also found differentially expressed in developing mouse cardiac tissue and in cellular proliferation.</p>
                                    <p>A DE assay performed with k=9 (See Supplshows that a major part of the population associated with AY036118 also has increased expression for  Gelsolin (<a href="http://www.informatics.jax.org/marker/MGI:95851">Gsn</a>),
                                        a protein associated with muscle cells and with a known role in hypertrophy and contraction in the heart. One possible explanation then could be that cells from cluster 4 might represent a subpopulation of stem cells that have both high potential for proliferation while also being able to differentiate into cardiomyocytes.</p>
                                    <p>The uncertainty and discrepancies observed during clustering and the biological assessment are to be expected up to a certain degree when doing unsupervised clustering, since it is mostly treated as a previous exploratory step before moving onto more powerful methods once more information on the population is available.
                                        One such case could be first identifying specific markers in biological assays that can then be used to pre-label the populations to perform semi-supervised or supervised clustering.</p>
                                    <hr class="major" />

									<h2 id="conclussion">6.2- Conclusions</h2>
                                    <p>Data analysis during this project has led to the conclusion that there might indeed be subpopulations in the PDGFRα+ve cells.</p>
                                    <p>Exploration of the data in dimensionally reduced spaces does not suggest any clear divisions in the population, supporting the notion first inferred during pre-processing that our data is quite homogenous.
                                        t-SNE and UMAP appear to be better performing DR techniques than the classical PCA. However, due to the stochastic nature of t-SNE with no random seed fixed, it produced less consistent graphs when compared to UMAP.</p>
                                    <p>Unsupervised clustering in SC3 does not allow us to affirm a definite number of subpopulations. Clustering suggests that there might be between 3 and 9 (surely an overestimation) of them, with most of our work performed under the assumption that there are 4 subpopulations.</p>
                                    <p>A study of the differentially expressed genes allows us to characterise what appears to be a subpopulation of cells with higher expression of AY036118, suggesting that the cells from cluster 4 (and in lesser degree 3) could have increased proliferative and cardiogenic properties within in the PDGFRα+ve cells.
                                        The other two possible subpopulations might represent cells in a more differentiated state towards muscle cells, or maybe just with higher metabolic activities, for they show increased expression of mitochondrial, ribosomal and possibly muscle-related markers if we further increase resolution by choosing a k of 9 (see <a href="supplementary.html">Supplementary Material)</a>.</p>
                                    
                                    <hr class="major" />
                                    
                                    <h2 id="future">6.3- Future work</h2>
                                    <p>As more traditional visualisation and clustering methods do not render extremely satisfactory results, it might be worth exploring the idea of applying radically different approaches such as pseudotime analysis or diffusion maps.
                                        This kind of approaches have been built to identify, in a single population, different stages along a process (such as differentiation), and single cell tailored R toolkits are already available (<a href="http://cole-trapnell-lab.github.io/monocle-release/">Monocle</a>).</p>
                                    <p>One interesting observation when looking at the marker genes found after clustering is that Serpinf1 is known to play a role in the endocrine system, inhibiting proliferation in pancreas and the prostate. Although surely only a curious coincidence,
                                        it would be interesting to study the presence of markers on a secretory phenotype in our population, since paracrine effects have been postulated as a possible mechanism for CPC regenerative role (<a href="#">Li et al., 2012</a>; <a href="#">Noseda et al., 2015</a>).</p>
									<hr class="major" />
									<p align="right"><a href="refs.html" class="button primary">Next</a></p>

								</section>

						</div>
					</div>

				<!-- Sidebar -->
				<div id="sidebar">
					<div class="inner">

						<!-- Menu -->
						<nav id="menu">
								<header class="major">
									<h2>Menu</h2>
								</header>
								<ul>
									<li><a href="index.html">Abstract</a></li>
									<li><a href="background.html">1.- Background</a></li>
									<li><a href="aim.html">2.- Aims and Initial Data</a></li>
									<li><a href="intro.html">3.- Introduction and pipeline</a></li>
									<li>
										<span class="opener">4.- Pre-processing</span>
										<ul>
											<li><a href="pre-process.html">4.1- QC and filtering</a></li>
											<li><a href="pre-process.html#norm">4.2- Normalisation</a></li>
										</ul>
									</li>
									<li>
										<span class="opener">5.- Visualisation and Clustering</span>
										<ul>
											<li><a href="analysis.html">5.1- Dimensionality Reduction</a></li>
											<li><a href="analysis.html#clust">5.2- Unsupervised clustering</a></li>
											<li><a href="analysis.html#bio">5.3- Biological analysis</a></li>
										</ul>
									</li>
									<li>
										<span class="opener">6.-Discussion</span>
										<ul>
											<li><a href="discussion.html">6.1- Discussion</a></li>
											<li><a href="discussion.html#conclussion">6.2- Conclusions</a></li>
											<li><a href="discussion.html#future">6.3- Future work</a></li>
										</ul>
									</li>
									<li><a href="refs.html">7.-References</a></li>
									<li><a href="supplementary.html">Supplementary Material</a></li>
								</ul>
							</nav>

						<!-- Section -->
							<!-- <section>
								<header class="major">
									<h2>Ante interdum</h2>
								</header>
								<div class="mini-posts">
									<article>
										<a href="#" class="image"><img src="images/pic07.jpg" alt="" /></a>
										<p>Aenean ornare velit lacus, ac varius enim lorem ullamcorper dolore aliquam.</p>
									</article>
									<article>
										<a href="#" class="image"><img src="images/pic08.jpg" alt="" /></a>
										<p>Aenean ornare velit lacus, ac varius enim lorem ullamcorper dolore aliquam.</p>
									</article>
									<article>
										<a href="#" class="image"><img src="images/pic09.jpg" alt="" /></a>
										<p>Aenean ornare velit lacus, ac varius enim lorem ullamcorper dolore aliquam.</p>
									</article>
								</div>
								<ul class="actions">
									<li><a href="#" class="button">More</a></li>
								</ul>
							</section> -->

						<!-- Section -->
						<section>
							<header class="major">
								<h2>Acknowledgements</h2>
							</header>
							<p>I would like to thank Prof. Mauricio Barahona and Dr. Michela Noseda for their supervision and guidance throughout the project.<br />Special thanks as well to Andrea Massaia, PhD, for all his support and scientific insights that made this project possible. </p>

						</section>

					<!-- Footer -->
						<footer id="footer">
							<p class="copyright">&copy; Ferran Cardoso.<br />Project 2: Data Analysis. <br />MSc in Bioinformatics and Theoretical Systems Biology.<br />Imperial College London. <br /> Design: <a href="https://html5up.net">HTML5 UP</a>.</p>
						</footer>

					</div>
				</div>

			</div>

		<!-- Scripts -->
			<script src="assets/js/jquery.min.js"></script>
			<script src="assets/js/browser.min.js"></script>
			<script src="assets/js/breakpoints.min.js"></script>
			<script src="assets/js/util.js"></script>
			<script src="assets/js/main.js"></script>

	</body>
</html>